core/asset/asset.go

```diff
@@ -87,8 +85,8 @@ func (reg *Registry) Define(ctx context.Context, xpubs []string, quorum int, def
 	asset := &Asset{
 		Definition:       definition,
 		IssuanceProgram:  issuanceProgram,
-		InitialBlockHash: reg.initialBlockHash,
-		AssetID:          bc.ComputeAssetID(issuanceProgram, reg.initialBlockHash, 1),
+		InitialBlockHash: reg.chain.InitialBlockHash,
```

Comment:
If we read `reg.chain.InitialBlockHash` like this, then I think we can't updated `reg.chain.InitialBlockHash` on the first block committed because concurrent read/writes are possible. Maybe we should pass in the blockchain ID to `protocol.NewChain` explicitly.